src/lib.rs

```diff
@@ -262,4 +262,57 @@ mod tests {
             assert_eq!(case.want, parse_line(case.line));
         }
     }
+
+    #[test]
+    fn test_results_sort_correctly() {
+        let mut results = vec![
+            TestResult {
+                module: Some("zeta".into()),
+                name: "zebra".into(),
+                status: Status::Pass,
+            },
+            TestResult {
+                module: None,
+                name: "alpha".into(),
+                status: Status::Fail,
+            },
+            TestResult {
+                module: Some("alpha".into()),
+                name: "beta".into(),
+                status: Status::Ignored,
+            },
+            TestResult {
+                module: Some("alpha".into()),
+                name: "alpha".into(),
+                status: Status::Pass,
+            },
+        ];
+
+        results.sort();
+
+        let expected = vec![
+            TestResult {
+                module: None,
+                name: "alpha".into(),
+                status: Status::Fail,
+            },
+            TestResult {
+                module: Some("alpha".into()),
+                name: "alpha".into(),
+                status: Status::Pass,
+            },
+            TestResult {
+                module: Some("alpha".into()),
+                name: "beta".into(),
+                status: Status::Ignored,
+            },
+            TestResult {
+                module: Some("zeta".into()),
+                name: "zebra".into(),
+                status: Status::Pass,
+            },
+        ];
+
+        assert_eq!(results, expected);
```

Comment:
When tests have multiple assertions, it's not always easy to see which one is failing. It's helpful to add a message:

```suggestion
        assert_eq!(results, expected, "wrong order");
```